community/CONTRIBUTING.md

```diff
@@ -5,6 +5,192 @@ Thanks for considering contributing to OPEA project. The contribution process is
+    ├── dataprep
+    ├── embeddings
+    ├── feedback_management
+    ├── finetuning
+    ├── guardrails
+    ├── intent_detection
+    ├── llms
+    ├── lvms
+    ├── nginx
+    ├── prompt_registry
+    ├── reranks
+    ├── retrievers
+    ├── tts
+    ├── vectorstores
+    └── web_retrievers
+```
+
+2. Follow the folder structure in the TEI embedding component below:
+
+```
+GenAIComps
+├── comps
+│   └── embeddings
+│       ├── __init__.py
+│       └── tei     #vendor name or serving framework name
+│           ├── langchain
+│           │   ├── Dockerfile        
+│           │   ├── Dockerfile.amd_gpu         
+│           │   ├── Dockerfile.nvidia_gpu      
+│           │   ├── embedding_tei.py    # definition and registration of microservice
+│           │   ├── README.md
+│           │   └── requirements.txt
+│           └── llama_index
+│               └── . . .
+├── tests
+│   └── embeddings
+│       ├── test_embeddings_tei_langchain.sh
+│       ├── test_embeddings_tei_langchain_on_amd_gpu.sh     
+│       └── test_embeddings_tei_llama_index.sh
+└── README.md
+
+```
+- **File Descriptions**:
+  - `embedding_tei.py`: This file defines and registers the microservice. It serves as the entrypoint of the Docker container. Refer to [whisper ASR](https://github.com/opea-project/GenAIComps/tree/main/comps/asr/whisper) for a simple example or [TGI](https://github.com/opea-project/GenAIComps/blob/main/comps/llms/text-generation/tgi/llm.py) for a more complex example that required adapting to OpenAI API.
+  - `requirements.txt`: This file is used by Docker to install the necessary dependencies.
+  - `Dockerfile`: Used to generate the service container image. Please follow naming conventions:
+    - Dockerfile: `Dockerfile.[vendor]_[hardware]`, vendor and hardware in lower case (i,e Dockerfile.amd_gpu)
+    - Docker Image: `opea/[microservice type]-[microservice sub type]-[library name]-[vendor]-[hardware]:latest` all lower case (i,e opea/llm-vllm-intel-hpu, opea/llm-faqgen-tgi-intel-hpu-svc)
+
+  - `tests/[microservices type]/` : contains end-to-end test for microservices Please refer to an example [test_asr_whisper.sh](https://github.com/opea-project/GenAIComps/blob/main/tests/asr/test_asr_whisper.sh). Please follow naming convention:`test_[microservice type]_[microservice sub type]_[library name]_on_[vendor]_[hardware].sh` 
+  - `tests/cores/` : contains UT test for core python components (orchestrator, gateway...). Please follow naming convention:`test_[core component].sh` 
+
+  - `README.md`: at minimum it should include: microservice description, build and start microservice commands and curl command with expected output.
+
+4. Now you have created all the required files, and validated your service. Last step is to modify the `README.md` at the component level `GenaIComps/comp/microservice type` to list your new component. Now you are ready to file your PR!. In the upcoming release maintainers will push the Docker Image to the Docker Hub. 
```

Comment:
at the component level `GenaIComps/comp/microservice type` 

should be 

at the component level `GenaIComps/comps/microservice type`

Reply:
done